redash/models.py

```diff
@@ -790,6 +790,8 @@ def store_result(cls, org, data_source, query_hash, query, data, run_time, retri
             Query.data_source == data_source)
         for q in queries:
             q.latest_query_data = query_result
+            # ugly hack to avoid Query's updated_at from changing in this specific scenario
+            q.updated_at = q.updated_at + datetime.timedelta(milliseconds=1)
```

Comment:
Let's not do that and instead fix the `TimestampMixin` class to optionally support not updating the `updated_at` value.

To do that we first have to change the mixin to stop using `onupdate` since it doesn't provide as with the hooks AFAIK to make educated guess, e.g. we don't have access to the ORM instance.

Instead we can use the [`before_update` ORM event](https://docs.sqlalchemy.org/en/latest/orm/events.html#sqlalchemy.orm.events.MapperEvents.before_update) to conditionally update the `updated_at` column:

```python
from sqlalchemy.event import listens_for

class TimestampMixin(object):
    updated_at = Column(db.DateTime(True), default=utcnow, nullable=False)
    created_at = Column(db.DateTime(True), default=utcnow, nullable=False)

@listens_for(TimestampMixin, 'before_update', propagate=True)
def timestamp_before_update(mapper, connection, target):
    # Check if we really want to update the updated_at value
    if hasattr(target.skip_updated_at):
        return
    target.updated_at = utcnow()
```

Then inside the `store_result` loop you can simply do:

```python
q.skip_updated_at = True
```

Reply:
@washort Just to be sure, do you know a reason not to use the event system for the timestamp mixin by any chance?

Reply:
Thanks @jezdez, looks a lot better 👍